lib/round-service.js

```diff
@@ -0,0 +1,185 @@
+/** @typedef {{id: number; start_time: string; end_time: string; max_tasks_per_node: number; }} Round */
+/** @typedef {{ roundDurationMs: number; checkRoundIntervalMs: number }} RoundConfig */
+
+export class RoundService {
+  /**
+   * @type {Round | null}
+   */
+  #currentRound = null
+  #isInitializing = false
+  /**
+   * @type {NodeJS.Timeout | null}
+   */
+  #checkRoundIntervalId = null
+  #db
+  #config
+  #taskingService
+
+  /**
+   * @param {import('./typings.js').PgPool} db
+   * @param {import('./tasking-service.js').TaskingService} taskingService
+   * @param {RoundConfig} config
+   */
+  constructor (db, taskingService, config) {
+    this.#db = db
+    this.#config = config
+    this.#taskingService = taskingService
+  }
+
+  /**
+   * Start the round service
+   */
+  async start () {
+    if (this.#isInitializing) return
+    this.#isInitializing = true
+
+    try {
+      await this.#initializeRound()
+      this.#scheduleRoundCheck()
+      console.log(`Round service started. Round duration: ${this.#config.roundDurationMs / 60000} minutes`)
+    } catch (error) {
+      console.error('Failed to start round service:', error)
```

Comment:
```suggestion
      console.error('Failed to start round service:', error)
      throw err
```

We don't want to ignore this error, and make it crash the process